distribution/lib/Standard/Base/0.2.26-SNAPSHOT/src/System/File.enso

```diff
@@ -727,3 +848,28 @@ type File_Error
         Access_Denied_Error file -> "You do not have permission to access the file at " + file.path + "."
         Io_Error msg -> "An IO error has occurred: " + msg.to_text + "."
 
+
+## PRIVATE
+
+   Utility function that lists immediate children of a directory.
+list_immediate_children directory =
+    arr = directory.prim_file.list
+    Vector.new (Polyglot.get_array_size arr) ix->
+        File (arr.at ix)
+
+## PRIVATE
+
+   Utility function that returns all descendants of the provided file, including
+   that file itself. If the file is not a directory, a list containing only
+   itself is returned.
+list_descendants file =
```

Comment:
They may be private, but type signatures are good for sanity (this one and below).